LiteCore/Support/Logging.cc

```diff
@@ -60,6 +58,31 @@ namespace litecore {
     static ofstream *sFileOut = nullptr;
     static LogEncoder* sLogEncoder = nullptr;
     static mutex sLogMutex;
+    static const int kNumLogFiles = 10;
+
+    static void purgeOldLogs(const string& filePath)
+    {
+        FilePath filePathObj(filePath);
+        FilePath logDir = filePathObj.dir();
+        multimap<time_t, FilePath> logFiles;
+        const int magicNumber = *(int*)LogEncoder::kMagicNumber;
```

Comment:
You're assuming 'int' is 32-bit, which isn't necessarily true.
I think it would be cleaner to compare slices:
`slice(LogEncoder::kMagicNumber, 4) == slice(magicBuffer, 4)`

Reply:
OMG!  I've been in .NET for too long